test/v2/ThirdPartyRegistry.spec.js

```diff
@@ -4544,6 +4548,498 @@ describe('ThirdPartyRegistry', function () {
+      )
+      expect(tp.isApproved).to.be.equal(true)
+      expect(tp.consumedSlots).to.be.eq.BN(0)
+
+      const _getSignature = (qty) =>
+        getSignature(
+          thirdPartyRegistryContract,
+          thirdParty1[0],
+          qty,
+          dummyBytes32,
+          manager,
+          domain,
+          version
+        )
+
+      const sig1 = await _getSignature(qty1)
+      const sig2 = await _getSignature(qty2)
+      const sig3 = await _getSignature(qty3)
+
+      const { logs } =
+        await thirdPartyRegistryContract.reviewThirdPartyWithRoot(
+          thirdParty1[0],
+          dummyBytes32,
+          [
+            [qty1, dummyBytes32, sig1.r, sig1.s, sig1.v],
+            [qty2, dummyBytes32, sig2.r, sig2.s, sig2.v],
+            [qty3, dummyBytes32, sig3.r, sig3.s, sig3.v],
+          ],
+          fromCommitteeMember
+        )
+
+      tp = await thirdPartyRegistryContract.thirdParties(thirdParty1[0])
+
+      expect(tp.root).to.be.equal(dummyBytes32)
+      expect(tp.isApproved).to.be.equal(true)
+      expect(tp.consumedSlots).to.be.eq.BN(total)
+
+      expect(logs.length).to.be.equal(4)
+
+      const assertConsumeLog = (log, qty) => {
+        expect(log.event).to.be.equal('ItemSlotsConsumed')
+        expect(log.args._thirdPartyId).to.be.equal(thirdParty1[0])
+        expect(log.args._qty).to.be.eq.BN(qty)
+        expect(log.args._signer).to.be.equal(manager)
+        expect(log.args._sender).to.be.equal(committeeMember)
+      }
+
+      assertConsumeLog(logs[0], qty1)
+      assertConsumeLog(logs[1], qty2)
+      assertConsumeLog(logs[2], qty3)
+
+      expect(logs[3].event).to.be.equal('ThirdPartyReviewedWithRoot')
+      expect(logs[3].args._thirdPartyId).to.be.equal(thirdParty1[0])
+      expect(logs[3].args._root).to.be.eq.BN(dummyBytes32)
+      expect(logs[3].args._isApproved).to.be.equal(true)
+      expect(logs[3].args._sender).to.be.equal(committeeMember)
+    })
+
+    it('reverts when sender is not from commitee', async function () {
+      await assertRevert(
+        thirdPartyRegistryContract.reviewThirdPartyWithRoot(
+          thirdParty1[0],
+          dummyBytes32,
+          [],
+          fromUser
+        ),
+        'TPR#onlyCommittee: CALLER_IS_NOT_A_COMMITTEE_MEMBER'
+      )
+    })
+
+    it('reverts when third party is not registered', async function () {
+      await assertRevert(
+        thirdPartyRegistryContract.reviewThirdPartyWithRoot(
+          thirdParty2[0],
+          dummyBytes32,
+          [],
+          fromCommitteeMember
+        ),
+        'TPR#_checkThirdParty: INVALID_THIRD_PARTY'
+      )
+    })
+  })
+
+  describe('consumeSlots', function () {
+    beforeEach(async function () {
+      await thirdPartyRegistryContract.addThirdParties(
+        [thirdParty1],
+        fromThirdPartyAgregator
+      )
+    })
+
+    it('should do nothing if no consume slots params are provided', async function () {
```

Comment:
This test should be a revert. Sending a transaction that does nothing it not worth it

Reply:
Other functions are not validating empty arrays